test/event_test.py

```diff
@@ -834,6 +870,57 @@ def test_poll(self):
         self.assertEqual(pygame.event.poll().type, pygame.NOEVENT)
 
 
+class EventModuleTestsWithTiming(unittest.TestCase):
+    __tags__ = ["timing"]
+
+    def test_event_wait(self):
+        """Ensure wait() waits for an event on the queue."""
+        # Test case without timeout.
+        event = pygame.event.Event(EVENT_TYPES[0], **EVENT_TEST_PARAMS[EVENT_TYPES[0]])
+        pygame.event.post(event)
+        wait_event = pygame.event.wait()
+
+        self.assertEqual(wait_event.type, event.type)
+
+        # Test case with timeout and no event in the queue.
+        wait_event = pygame.event.wait(100)
+        self.assertEqual(wait_event.type, pygame.NOEVENT)
+
+        # Test case with timeout and an event in the queue.
+        event = pygame.event.Event(EVENT_TYPES[0], **EVENT_TEST_PARAMS[EVENT_TYPES[0]])
+        pygame.event.post(event)
+        wait_event = pygame.event.wait(100)
+
+        self.assertEqual(wait_event.type, event.type)
+
+        # test wait with timeout waits for the correct duration
+        pygame.time.set_timer(pygame.USEREVENT, 50, 3)
+
+        for wait_time, expected_type, expected_time in (
+            (60, pygame.USEREVENT, 50),
+            (65, pygame.USEREVENT, 50),
+            (20, pygame.NOEVENT, 20),
+            (45, pygame.USEREVENT, 30),
+            (70, pygame.NOEVENT, 70),
+        ):
+            start_time = time.perf_counter()
+            self.assertEqual(pygame.event.wait(wait_time).type, expected_type)
+            self.assertAlmostEqual(
+                time.perf_counter() - start_time, expected_time / 1000, delta=0.01
+            )
```

Comment:
I'd rather have the test have wait_time == expected_time, with a greater delta.

I don't like having a test that is like "oh, lets wait 60 ms for this, but we expect it to happen in 50ms, and if it took 61ms the test will fail."

Reply:
Well, we can't pass the exact expected time to `pygame.event.wait`, we have to give some delta for that function itself

Reply:
Why not? Just use the delta time in assertAlmostEqual.

Function calling overhead should be insignificant in the order of milliseconds, if that's what you mean. Also the wait vs expected gaps you have so far don't seem to be a consistent pattern.

Reply:
what time is to be passed to the `pygame.event.wait` then? If the exact duration is passed, the event can be missed due to timing inaccuracies. A delta time must be passed to both `event.wait` and the corresponding `assertAlmostEqual` time check.

I wrote this block to try to test a wider range of stuff, including cases like waiting for a bit longer, waiting a bit less and missing an event which is then caught in the next wait loop. That's why there's no consistent pattern here

Reply:
I understand this test is magic-number-ey, but I feel this is a good way to test timing related stuff

Reply:
Alrighty, discord discussion straightened me out. I thought you were posting them and checking to see that you received them in a good time, but it makes sense now I realize you're grabbing events posted by a timer.